content/purgatorio/now-go-build/transfer-learning.md

```diff
@@ -1,5 +1,111 @@
 ---
 title: Transfer Learning
 author: clone95
-description: ---
----
\ No newline at end of file
+description: A detailed guide about what is Transfer Learning, how to use it, and a lot of resources found on the web that shares pre-trained models.
+---
+
+# Index 
+
+- [What is Transfer Learning](#Why-Do-You-Need-Transfe-Learning)
+- [Transfer Learning Magical Properties](#[TransferLearning-Inner-Workings)
+- [How Does Transfer Learning Work](#How-Does-Transfer-Learning-Work)
+- [Where to Find Pretrained Models](#Where-to-Find-Pretrained-Models)
+- [Conclusions](#Conclusions)
+
+##  What is Transfer Learning
+
+[Transfer Learning](https://en.wikipedia.org/wiki/Transfer_learning) is a research problem in Machine Learning that focuses on storing knowledge gained while solving one problem and applying it to a different but related problem. 
+
+This is done by _pre-training_ a model on a dataset (usually bigger and with generic content, like [ImageNet](http://www.image-net.org/)), and then _fine-tuning_ the model on a smaller and similar dataset.
+
+For example, you can fine-tune a pre-trained model from ImageNet on your much smaller dataset of images, in order to benefit the pre-training and obtain a series of almost-magical properties and results.
+
+To get a comprehensive overview of Why, What and How is used Transfer Learning, plus possible applications, read this awesome article:
+
+- [Transfer Learning - Machine Learning's Next Frontier](https://ruder.io/transfer-learning/index.html)
+
+
+#### Transfer Learning Magical Properties
+
+The theory of Transfer Learning suggests that when using a pre-trained model on a similar dataset:
+
+**You need less**:
+- Data: you need much smaller datasets to obtain the same (or better) model performances
+- Computational Power / Computational Time: empirical results have shown that pre-trained models reach top performances after much fewer epochs of training, that starting the training from scratch
+
+**You improve:**
+- Out-of-sample generalization: when predicting on test data you often obtain better accuracy, thus reducing overfitting (or whatever metric you're using)
+- Robustness: you make the model more "robust" to real-world low-quality data  
+
+So, **the first thing to do** every time you frame a new problem, is to ask yourself:
+
```

Comment:
All of the above is true only if the base model (so-called pre-trained model) used for transfer learning is also sound and produces reliable results, is compatible, these are the pre-requisite.

Reply:
true I add these